### PR TITLE
feat: expose site feature flags to EF registry

### PR DESCRIPTION
**Pull request by eduardoboucas**
#### Summary

Feature flags coming from the site info are not being surfaced to the edge functions registry. This PR fixes that.

### CODE REVIEW

File: src/lib/edge-functions/proxy.ts [hunk 1]
```diff
@@ -136,11 +137,9 @@ export const initializeProxy = async ({
 }) => {
   const userFunctionsPath = config.build.edge_functions
   const isolatePort = await getAvailablePort()
-  const buildFeatureFlags = {
-    edge_functions_npm_modules: true,
-  }
   const runtimeFeatureFlags = ['edge_functions_bootstrap_failure_mode', 'edge_functions_bootstrap_populate_environment']
   const protocol = settings.https ? 'https' : 'http'
+  const buildFeatureFlags = { ...getFeatureFlagsFromSiteInfo(siteInfo), edge_functions_npm_modules: true }
```
Comment (eduardoboucas) [inline src/lib/edge-functions/proxy.ts:142]: This flag is not used anymore. We'll be able to get rid of this special case once we update `@netlify/edge-bundler`.

File: src/lib/edge-functions/registry.ts [hunk 2]
```diff
@@ -80,10 +80,11 @@ function traverseLocalDependencies(
 }
 
 export class EdgeFunctionsRegistry {
+  public importMapFromDeployConfig?: string
```
Comment (Skn0tt) [inline src/lib/edge-functions/registry.ts:83]: I don't see this referenced any where else in this diff. Assuming it's a type addition, no runtime effect.

**Review by Skn0tt — DISMISSED** [review]

File: src/lib/edge-functions/registry.ts (hunk 2 above)
Comment (eduardoboucas) [reply src/lib/edge-functions/registry.ts:83]: I moved it from L86 to separate private properties from public ones.

File: src/lib/edge-functions/registry.ts [hunk 3]
```diff
@@ -31,15 +32,14 @@ type RunIsolate = Awaited<ReturnType<typeof import('@netlify/edge-bundler').serv
 
 type ModuleJson = ModuleGraph['modules'][number]
 
-const featureFlags = { edge_functions_correct_order: true }
```
[the remark this answers is not in the document]
Comment (eduardoboucas) [reply src/lib/edge-functions/registry.ts:34]: No. https://github.com/netlify/edge-bundler/blob/fd07443937e6b1d3db4fca6a425e581bc791e5df/node/feature_flags.ts#L1-L3

**Review by Skn0tt — APPROVED** [review]

**Pull request merged**